sdk/core/azure-core-amqp/inc/azure/core/amqp/network/amqp_header_detect_transport.hpp

```diff
@@ -8,22 +8,21 @@
 
 namespace Azure { namespace Core { namespace Amqp { namespace Network { namespace _internal {
 
-  /** A transport that detects the AMQP header on an incoming message. Used when listening for
-   * incoming AMQP connections and messages.
+  /** @brief Factory to create a transport which detects the AMQP header on an incoming message.
+   * Used when listening for incoming AMQP connections and messages.
    */
-  class AmqpHeaderDetectTransport final : public Transport {
+  class AmqpHeaderDetectTransportFactory final {
```

Comment:
delete/move constructors and a destructor to private

Reply:
What constructors and destructors? There is a single Create method on the AmqpHeaderDetectTransportFactory.

Reply:
I replied just a bit below, implicitly generated public constructors and destructors.
```
    auto p = new AmqpHeaderDetectTransportFactory (); // compiles
```

Reply:
Ok, that makes more sense.